src/lib/parse.ts

```diff
@@ -11,7 +11,7 @@ type ParseResult = {
   date: Date | null
   missingPunctuation: string
 }
-export function parse(str: string, tokens: FormatToken[], baseDate?: Date): ParseResult {
+export function parse(str: string, tokens: FormatToken[], baseDate: Date | null): ParseResult {
```

Comment:
Haven't looked into this enough, but if `baseDate` is null, I'm not sure if a date should be returned

Reply:
Should be good :D